wdl-analysis/src/analyzer.rs

```diff
@@ -320,26 +366,31 @@ pub enum DocumentChange {
 ///
 /// Note that dropping the analyzer is a blocking operation as it will wait for
 /// the queue thread to join.
+///
+/// The type parameter is the context type passed to the progress callback.
 #[derive(Debug)]
-pub struct Analyzer {
+pub struct Analyzer<Context> {
     /// The sender for sending analysis requests to the queue.
-    sender: ManuallyDrop<mpsc::UnboundedSender<Request>>,
+    sender: ManuallyDrop<mpsc::UnboundedSender<Request<Context>>>,
```

Comment:
Perhaps it is simply that this is confined to the `wdl` crates, but the purpose of
introducing `Context` here is not clear to me. Could you expound briefly on why you
introduced that?

Reply:
I felt the `Option<String>` context argument that was used before was too specific to our LSP implementation, so instead changed it to a type parameter so that anyone using `wdl-analysis` can pass whatever they want to their progress callbacks (like in the case of the CLI, we can pass the progress bar in now, which I'll change to do).

Reply:
Okay I see now. Nevermind.